framework/web/Request.php

```diff
@@ -196,6 +243,34 @@ public function resolve()
     }
 
     /**
+     * Filters headers according to the trusted host config.
+     * @param array $headers
+     * @since 2.0.13
+     * @return array filtered headers.
+     */
+    protected function filterHeaders(array $headers)
+    {
+        $host = $this->getUserHost();
```

Comment:
okay, I see now, but its not getting it from a header in your implementation, its using the `$_SERVER` variable provided by the webserver. We have an inconsistency in behavior then:

- `getUserHost()` will only use the hostname provided by the webserver
- `getUserIp()` will use headers or fallback to `getRemoteIp()`

I would expect `getUserHost()` to return the host of the IP returned by `getUserIp()`, which is not the case if that one is taken from a header.

A corresponding `getRemoteHost()` would be preferable.